mipengine/controller/api/algorithm_request_dto.py

```diff
@@ -8,19 +9,28 @@
 USE_SMPC_FLAG = "smpc"
 
 
-class AlgorithmInputDataDTO(BaseModel):
+class ImmutableBaseModel(BaseModel, ABC):
+    class Config:
+        allow_mutation = False
+
+
+class AlgorithmInputDataDTO(ImmutableBaseModel):
     data_model: str
     datasets: List[str]
-    filters: dict = None
+    filters: Optional[dict]
     y: Optional[List[str]]
     x: Optional[List[str]]
 
     class Config:
         allow_mutation = False
 
 
+PARAMETERS_TYPE = Dict[str, Any]
```

Comment:
Simplifying the type definition.

Instead of `Optional[Dict[str, Dict[str, Any]]]` -> `Optional[Dict[str, PARAMETERS_TYPE]]`

Makes it a little bit more readable.